ws-nextjs-app/pages/[service]/send/[id]/FormContext/utils/validateFunctions.ts

```diff
@@ -108,15 +108,74 @@ const isValidTel: (data: FieldData) => FieldData = (data: FieldData) => {
   };
 };
 
-const isValidFile: (data: FieldData) => FieldData = (data: FieldData) => {
-  const { value, required, wasInvalid } = data;
+const isValidFiles: (data: FieldData) => FieldData = (data: FieldData) => {
+  const { value, required, wasInvalid, min, fileTypes, max } = data;
+
+  const MAX_PAYLOAD_SIZE = 1288490189;
+  const RESERVED_FORM_DATA_SIZE = 10000;
+  // we're making the assumption that each file chooser field is only allowed the max payload size less some for data (estimated maximum size ever needed)
+  // This will break if you have 2 file choosers in the form as each will have a separate allowance which could result in the total payload being exceeded (fails on Backend) - quite an edge case though
+  const TOTAL_FILES_MAX_SIZE = MAX_PAYLOAD_SIZE - RESERVED_FORM_DATA_SIZE;
 
   let isValid = true;
-  if ((value as File[])?.length === 0 && required) isValid = false;
+  let messageCode = null;
+  let hasNestedErrorLabel = false;
+
+  // CHECK INDIVIDUAL FILES
+  const validatedFiles = (value as FileData[]).map((fileData: FileData) => {
+    const { file } = fileData;
+    let fileMessageCode = null;
+
+    // Chrome interprets a wma file as video so the following checks whether this has happened and switches the video back to audio
+    const fileType =
+      file.type === 'video/x-ms-wma' ? 'audio/x-ms-wma' : file.type;
+
+    if (!fileTypes?.includes(fileType)) {
+      fileMessageCode = InvalidMessageCodes.WrongFileType;
+      hasNestedErrorLabel = true;
+      isValid = false;
+    }
+    if (file.size <= 0) {
+      fileMessageCode = InvalidMessageCodes.FileTooSmall;
+      hasNestedErrorLabel = true;
+      isValid = false;
+    }
+
+    return { ...fileData, messageCode: fileMessageCode };
+  });
+  if (min != null && (value as FileData[])?.length < min && required) {
+    isValid = false;
+    messageCode = InvalidMessageCodes.NotEnoughFiles;
+    hasNestedErrorLabel = false;
+  }
+
+  if (max != null && (value as FileData[])?.length > max) {
+    isValid = false;
+    messageCode = InvalidMessageCodes.TooManyFiles;
+    hasNestedErrorLabel = false;
+  }
+
+  const totalUploadedSize = (value as FileData[]).reduce(
+    (accumulated, currentValue) => accumulated + currentValue.file.size,
+    0,
+  );
+
+  if (totalUploadedSize >= TOTAL_FILES_MAX_SIZE) {
+    isValid = false;
+    messageCode = InvalidMessageCodes.FileTooBig;
+    hasNestedErrorLabel = false;
+  }
 
```

Comment:
May not be an issue, but do we need to bail out early with each of these validation checks? Just thinking of the scenario whereby someone uploads more than `max`, but also breaches the `MAX_SIZE` as well? They'll get the `MAX_SIZE` error as it will overwrite the previous `max` length check.

Not really sure what should happen tbh, just a thought.

Reply:
We asked wendy about the UX of if theres more than one error, and she thinks we should just show one and we dont have a priority list for these. So i think maybe this is fine? e.g. they upload more than max **and** breach max size, they get shown the max size error. Then when they fix that, after submitting again if the more than max error is still an issue they get shown that.

Reply:
No probs, thanks for clarifying!